### ISSUE TITLE
memory datastore config setting would be nice

### ISSUE THREAD

**Issue by whyrusleeping** [ipfs/go-ipfs#1366]
We used to be able to specify in the config to use an 'in memory' datastore. This would be nice to have again for testing various things.

**Comment by jbenet** [comment]
:+1:

**Comment by whyrusleeping** [comment]
I have some basic code for this here: https://github.com/ipfs/go-ipfs/tree/feat/config-ds

But its probably best to wait on the s3 datastore changes.

**Comment by Stebalien** [comment]
We now have support for custom datastore configs and plugins. This should now be pretty trivial to implement if anyone wants to pick it up. You may find the following example useful: https://github.com/ipfs/go-ipfs-example-plugin/

**Comment by cfalcondair** [comment]
I'll have a crack.

**Comment by Stebalien** [comment]
Actually, it looks like we support this without any additional plugins. We just need to create a config profile for the "mem" datastore: https://github.com/ipfs/go-ipfs-config/blob/master/profile.go.

**Comment by magik6k** [comment]
Note that using memory datastore may break between restarts in weird ways:
* if used for entire datastore, it probably won't even start because of missing pinroot/filesroot
* if only used for blocks, pinning will not be happy. MFS may also break

(this should be fixable, but we have to be careful wmen fixing pinning)

**Comment by Jorropo** [comment]
There is support here:

https://github.com/ipfs/kubo/blob/afa891b0edf13333580e8b8a2aba6be624d63ad9/repo/fsrepo/datastores.go#L61

I'll close this given I don't see anything left to do, can reopen if I missed something obviously.
